### PR TITLE
feat(discord): discord service OAuth2

### LINKED ISSUE #77: Service Discord
add Discord service to use his api, so search how connect with OAuth 2.0

### PR DESCRIPTION
**Pull request by Benjicatch**
- close #77 
- connection to discord api
- update readme for .env
- update docker compose with environment

### CODE REVIEW

File: server/src/discord/discord.service.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import { PrismaService } from '../prisma/prisma.service';
+import { ServiceType } from '@prisma/client';
+
+@Injectable()
+export class DiscordService {
+  constructor(private prisma: PrismaService) {}
+  async discordLogin(@Res() req: any) {
+    const url =
+      'https://discord.com/api/oauth2/authorize?client_id=1189230397108793416&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A8080%2Fauth%2Fdiscord%2Fcallback&scope=identify+email';
+    req.redirect(url);
```
Comment (Pablodeibar) [inline server/src/discord/discord.service.ts:13]: Here we need a strategy caus the client id is exposed

**Review by Pablodeibar — CHANGES_REQUESTED** [review]

File: server/src/discord/discord.service.ts (hunk 1 above)
Comment (Benjicatch) [reply server/src/discord/discord.service.ts]: changed

**Review by Pablodeibar — DISMISSED** [review]
looks good

**Review by AlexGuillard — APPROVED** [review]
ok

**Review by Pablodeibar — APPROVED** [review]
looks good

**Review by Pablodeibar — APPROVED** [review]
looks good

**Review by AlexGuillard — APPROVED** [review]
ok

**Review by AlexGuillard — APPROVED** [review]
ok

**Pull request merged**